sendgrid/sendgrid.py

```diff
@@ -58,3 +57,14 @@ def api_key(self, value):
     @property
     def impersonate_subuser(self):
         return self._impersonate_subuser
+
+
+class SendGridClient(SendGridAPIClient):
+    def __init__(self, **opts):
+        super().__init__(**opts)
+
+    def send_mail(self, msg):
+        try:
+            return self.client.mail.send.post(request_body=msg.get())
+        except Exception as e:

```

Comment:
We can actually. The end user is concerned with the SendGrid module. So internally, we can use the `urllib` class `HTTPError` to catch exception and then raise a `SendGridException` in its place. So the end user will only see `SendGridException` whenever an error occurs. They need not worry about the internal use of urllib.